### PR TITLE
Add more layout rules

### PR DESCRIPTION
**Pull request by max-si-m**
Problem:
1. We always add space after frozen string literal manually, usually requested on code review. Added rule to fix that [Layout/EmptyLineAfterMagicComment](https://docs.rubocop.org/rubocop/cops_layout.html#layoutemptylineaftermagiccomment)
2. Spaces inside blocks and lambdas quite often inconsistent, and it is painful for me. Added rule to fix that [Layout/SpaceInsideBlockBraces](https://docs.rubocop.org/rubocop/cops_layout.html#layoutspaceinsideblockbraces)
3. Nice to have, trailing dot for methods [Layout/DotPosition](https://docs.rubocop.org/rubocop/cops_layout.html#layoutdotposition)

`demo.rb`
~~~ruby
# frozen_string_literal: true
class A
  def test
    "Dummy"
  end
end

# we have a lot of examples like this
->(x) {x + 1 }
proc {|x| x + 1 }

# I would preferr to have trailing dot as it easir to read and use in console
[1, 2, 3]
  .sum
~~~

Rubocop output:
~~~bash
$ rubocop demo.rb
Inspecting 1 file
W

Offenses:

demo.rb:2:1: C: [Correctable] Add an empty line after magic comments.
class A
^
demo.rb:9:8: C: [Correctable] Space missing inside {.
->(x) {x + 1 }
       ^
demo.rb:10:6: C: [Correctable] Space between { and | missing.
proc {|x| x + 1 }
     ^^
demo.rb:14:3: C: [Correctable] Place the . on the previous line, together with the method call receiver.
  .sum
  ^
~~~

### CODE REVIEW

**Comment by zhuravel** [comment]
1. > We always add space after frozen string literal manually

I prefer no new line. We have more places without a new line:

`"# frozen_string_literal: true\nclass"` 47 matches
`"# frozen_string_literal: true\n\nclass"` 40 matches

2. > Spaces inside blocks and lambdas quite often inconsistent

I’d prefer to add `SpaceBeforeBlockParameters: false` option:

```ruby
[1, 2, 3].each { |n| n * 2 } # bad
[1, 2, 3].each {|n| n * 2 } # good
```

I see absolutely no value in space between `{` and `|`.
But I will accept this rule if the magority wants to have a space there.

3. > I would prefer to have trailing dot

Good suggestion, always preferred this as it copy-pastes to irb with no issues unlike with a leading dot.

**Review by Doimhneacht — COMMENTED** [review]
I really dislike the lack of space `{|x|... }`, so I'm voting for adding the rule Maksym proposed.
Love the trailing dot, too.

**Comment by zhuravel** [comment]
@Doimhneacht 


Question is: Wdyt of `SpaceBeforeBlockParameters: false` option?

**Comment by Doimhneacht** [comment]
> SpaceBeforeBlockParameters: false

@zhuravel Sorry, I meant `{|x| ... }`. I don't like when there is no space between the brace and the block arg pipe.

File: config/layout.yml [hunk 1]
```diff
@@ -18,10 +18,15 @@ Layout/ConditionPosition:
...
   Enabled: true
+Layout/DotPosition:
+  Enabled: true
+  EnforcedStyle: trailing
 Layout/EmptyComment:
   Enabled: true
 Layout/EmptyLineBetweenDefs:
   Enabled: true
+Layout/EmptyLineAfterMagicComment:
+  Enabled: true
```
Comment (zhuravel) [inline config/layout.yml:29]: Let’s remove this cop. Other look good.

**Review by zhuravel — APPROVED** [review]

**Review by zhuravel — APPROVED** [review]

**Pull request merged**